rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt

```diff
@@ -1094,7 +1094,29 @@ data class DataStructValue(var value: String, private val optionalExternalLen: I
         } else if (data.declaredArrayInLine != null) {
             ProjectedArrayValue.forData(this, data)
         } else {
-            coerce(this.getSubstring(data.startOffset, data.endOffset), data.type)
+            val substring = this.getSubstring(data.startOffset, data.endOffset)
```

Comment:
I don't undestand what issue it fixes, can you make me an example?

Reply:
This fixes these cases:
| Value * | DS Field Type | Precision (Scale) | Result   | Notes |
| ---------------------- | -------------- | ----------------- | ------ | ------ |
| `'00520'` | `ZONED` | 5 (0) | 520 | |
| `'00520'` | `ZONED` | 5 (2) | 5.2 | |
| `'0052 '` | `ZONED` | | Error | The value must contain only numbers. |

`data` must be `NumberType` and the syntax have to respect the rule. In this case only for "without blank characters at the end". 
You might ask me: "why this logic is not in `coerce`? Because is a rule only for DS. First of this implementation I put this logic in `coerce` and some tests failed; in other condition, out from DS, is possible to have blank at the end. In `coerce` I don't know who is the parent of the type; I know only the value and the destination type.